docs/proposals/05-etcd-operator-tasks.md

```diff
@@ -0,0 +1,383 @@
+* If multiple tasks are invoked simultaneously or in `pending` state, then they will be executed in a First-In-First-Out (FIFO) manner.
+
+> Note: Dependent ordering among tasks will be addressed later which may enable concurrent execution of tasks.
+
+#### Deletion
+
+Upon completion of the task, whether it has failed, succeeded, or been rejected, Etcd-druid will ensure the garbage collection of the custom resources and any other Kubernetes resources created to execute the task. This will be done according to the `.spec.ttlSecondsAfterFinished` if defined in the [spec](#spec), or a default expiry time will be assumed.
+
+### Use Cases
+
+#### Recovery from permanent quorum loss
+
+Currently, identification and recovery from permanent quorum loss is done manually. The current proposal keeps the identification as manual, requiring a human operator to intervene and confirm that there is indeed a permanent quorum loss with no auto healing possibility. Once that is established then the next step to recover the cluster is now proposed to be automated. Recovery today is a [multi-step process](https://github.com/gardener/etcd-druid/blob/master/docs/operations/recovery-from-permanent-quorum-loss-in-etcd-cluster.md) and needs to be done carefully. Automation would ensure that we eliminate any errors from an operator.
+
+##### Task Config
+
+We do not need any config for this task. When creating an instance of `EtcdOperatorTask` for this scenario, `.spec.config` will be set to nil (unset).
+
+##### Pre-Conditions
+
+* There should be a quorum loss in a multi-member etcd cluster. For a single-member etcd cluster, invoking this task is unnecessary as the restoration of the single member is automatically handled by the backup-restore process.
+* There should not already be a permanent-quorum-loss-recovery-task running for the same etcd cluster.
+
+#### Trigger on-demand snapshot compaction
+
+`Etcd-druid` provides a configurable [etcd-events-threshold](https://github.com/gardener/etcd-druid/blob/master/docs/proposals/02-snapshot-compaction.md#druid-flags) flag. When this threshold is breached, then a [snapshot compaction](https://github.com/gardener/etcd-druid/blob/master/docs/proposals/02-snapshot-compaction.md) is triggered for the etcd cluster. However, there are scenarios where an ad-hoc snapshot compaction may be required.
+
+##### Possible scenarios
+
+* Full snapshot taken via compaction job is better as compared to full snapshot taken directly from etcd cluster as full snapshot taken via snapshot compaction is taken after [defragmentation](https://etcd.io/docs/v3.2/op-guide/maintenance/#defragmentation) which makes it smaller (more compacted) in size.
+* Full snapshot triggered on etcd cluster might cause load on running etcd cluster but the full snapshot taken by the compaction job is designed to run separately, thereby having no impact on the operational cluster.
```

Comment:
The points mentioned here aren't really *possible scenarios* in which an out-of-band snapshot compaction would be useful. They explain what an out-of-band snapshot compaction would do, and why they would be useful. "Why" is explained, but "Where" isn't.
This section could either be moved out of possible scenarios, or more cases where an out-of-band snapshot compaction would be useful can be written.

Reply:
agreed, I have now removed these points and added a new points.

Reply:
The snapshot integrity verification is a great scenario. Thanks!